src/lib.rs

```diff
@@ -464,8 +465,9 @@ pub fn check_transient_override(buffer: &mut [u8; 32]) -> Option<SlotId> {
         None
     };
 
-    if choice.is_some() {
+    if let Some(slot) = choice {
         buffer.fill(0);
+        buffer[0] = if slot == SlotId::A { 0 } else { 1 };
```

Comment:
In order for Hubris-now to know what Hubris-then set the transient request to, Hubris then has to put PREFER_SLOT_A, PREFER_SLOT_B or something not equal to either of those or zeros or zeros+1.
If the not-any-of-those-values survives, then Hubris-then's "message of didn't use the feature" gets passed through.
If all-zeros or all-zeros+1 is seen, then Hubris-then might have used the feature or maybe no Hubris instance since long ago has cleaned up.
This can be solved in Hubris pre-kernel by harvesting the value and setting that area to some constant.
However, if anyone is zeroing or otherwise initializing the transient override buffer, then the reading can still be ambiguous.

I would prefer that the transient preference buffer always be written with one of three values:
  - all zeros + 2 indicates that there had been no transient preference expressed.
  - all zeros indicates SlotId::A was preferred
  - all zeros + 1 indicates SlotId::B was preferred.
...or something similar. The leading zeros can be reserved by bootleby and ignored by stage1/Hubris until defined. This is the way.